packages/core/src/api/interactions.ts

```diff
@@ -429,4 +429,54 @@ export class InteractionsAPI {
 
 		return with_response ? response : undefined;
 	}
+
+	/**
+	 * Launches an activity and returns an interaction callback object
+	 *
+	 * @see {@link https://discord.com/developers/docs/interactions/receiving-and-responding#create-interaction-response}
+	 * @param interactionId - The id of the interaction
+	 * @param interactionToken - The token of the interaction
+	 * @param body - The callback data for launching the activity
+	 * @param options - The options for launching the activity
+	 */
+	public async launchActivity(
+		interactionId: Snowflake,
+		interactionToken: string,
+		body?: RESTPostAPIInteractionCallbackQuery & { with_response: true },
+		options?: Pick<RequestData, 'signal'>,
+	): Promise<RESTPostAPIInteractionCallbackWithResponseResult>;
+
+	/**
+	 * Launches an activity
+	 *
+	 * @see {@link https://discord.com/developers/docs/interactions/receiving-and-responding#create-interaction-response}
+	 * @param interactionId - The id of the interaction
+	 * @param interactionToken - The token of the interaction
+	 * @param body - The callback data for launching the activity
+	 * @param options - The options for launching the activity
+	 */
+	public async launchActivity(
+		interactionId: Snowflake,
+		interactionToken: string,
+		body?: RESTPostAPIInteractionCallbackQuery & { with_response?: false },
+		options?: Pick<RequestData, 'signal'>,
+	): Promise<undefined>;
```

Comment:
This overload will be selected when passing a `RESTPostAPIInteractionCallbackQuery` where `with_response` is `boolean`.
This could be a `true` in runtime and then return resolve to a `RESTPostAPIInteractionCallbackWithResponseResult`.

You could add another overload with `{ with_response: false }` where the function resolves to `undefined` and then the catch-all with an union of both.

Reply:
Adding onto this, it could prove useful to add tests into https://github.com/discordjs/discord.js/blob/main/packages/core/__tests__/types.test.ts.